### PR TITLE
Documentation for Fuseki Modules

### PR DESCRIPTION
**Pull request by afs**

### CODE REVIEW

File: source/documentation/fuseki2/fuseki-modules.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
+---
+title: Fuseki Modules
+---
+
+<em>Fuseki modules</em> are a mechniam to include extension code into a Fuseki
+server. Modules are invoked during the process of building a [Fuseki
+Main](./fuseki-main.html) server. The module can modify the server
+configuration, add new functiomnality, or react to a server being built and
+started.
+
```
Comment (rvesse) [inline source/documentation/fuseki2/fuseki-modules.md:10]: As a brand new feature may want an explicit note about the version when support was added for this

File: source/documentation/fuseki2/fuseki-modules.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+* `serverStopped` - called as the server stop 
+  (note, this is not always called because a server can simply exit the JVM).
+
+A Fuseki module does not need to implement all these steps, the default for all
+of them is "do nothing". Usually, an extension will only be interested in
+certain stpes, like the configuration and registry information of
+`configuration`.
+
+During the configuration step, the Fuskei configuration file for the server is
+available. If the server is built prorgammatically without a configuration file,
```
Comment (rvesse) [inline source/documentation/fuseki2/fuseki-modules.md:51]: Typo in programmatically

**Review by rvesse — APPROVED** [review]
LGTM, couple of minor comments

File: source/documentation/fuseki2/fuseki-modules.md (hunk 2 above)
Comment (afs) [reply source/documentation/fuseki2/fuseki-modules.md:51]: Fixed.

File: source/documentation/fuseki2/fuseki-modules.md (hunk 1 above)
Comment (afs) [reply source/documentation/fuseki2/fuseki-modules.md:10]: Done.

(Personal note - I don't mind such notes even if it does hint at "support" for older versions. We have quite good at compatibility so upgrading to a recent version is better. Even better, if we could run support branches and releases. "**_IF_**" ...

**Review by rvesse — APPROVED** [review]

**Pull request merged**